src/TableOfContents/index.tsx

```diff
@@ -128,6 +129,21 @@ function TableOfContentsInner<T extends TableOfContentsItem = TableOfContentsIte
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [contents, contents.length]);
 
+  // expand items marked as initially expanded and its children
+  React.useEffect(() => {
+    const itemsToExpand = contents.filter(item => item.startExpanded);
+    const childrenToExpand = flatMap(itemsToExpand, item =>
+      findDescendantIndices(item.depth ?? 0, contents.indexOf(item), contents.slice(contents.indexOf(item) + 1)),
+    );
+
+    setExpanded(current => ({
+      ...current,
+      ...Object.fromEntries(itemsToExpand.map(index => [index, true])),
+      ...Object.fromEntries(childrenToExpand.map(index => [index, true])),
+    }));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
I don't buy why a prop called `startExpanded` would make children expanded.

Also is that required to adhere to the spec? It only wants you to expand individual APIs, not groups within, so that should be enough.

Reply:
And instead of hacking a `useEffect`, you could move the logic to line 108:
```ts
  const [expanded, setExpanded] = React.useState({});
```
and replace the empty initializer

Reply:
I understood `Apis should start expanded` as fully expanding its content with groups, but it might be just first level depth.
If we don't need to expand children then it will be simpler and without useEffect

Reply:
I don't think you need it all the way, that can get very messy quick

Reply:
Updated to be less greedy when expanding